### PR TITLE
Update env-vars.md

### PR DESCRIPTION
**Pull request by HennaAbbas**
The first letter of a env var can be a underscore. Tested this locally and it works to set an env var to "_jhbdbf"

# Description
A brief description of the changes.

First letter can be a _. Aka "The first character of each environment variable must be a letter." is not valid. 

# Reasons
A link to a GitHub and/or JIRA issue (if applicable).
Otherwise, a brief sentence about why you made these changes.

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: [https://circleci.com/docs/style/style-guide-overview](https://circleci.com/docs/style/style-guide-overview).

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [ ] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

File: jekyll/_cci2/env-vars.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ Private environment variables enable you to store secrets safely, even when your
 ## Environment variable usage options
 {: #environment-variable-usage-options }
 
-CircleCI uses Bash, which follows the POSIX naming convention for environment variables. Valid characters include letters (uppercase and lowercase), digits, and the underscore. The first character of each environment variable must be a letter.
+CircleCI uses Bash, which follows the POSIX naming convention for environment variables. Valid characters include letters (uppercase and lowercase), digits, and the underscore.
```
Comment (jtatarik) [inline jekyll/_cci2/env-vars.md:46]: This is correct, but incomplete. Quoting from https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap08.html:

```
Environment variable names used by the utilities in the Shell and Utilities volume of POSIX.1-2017 consist solely of uppercase letters, digits, and the <underscore> ( '_' ) from the characters defined in [Portable Character Set](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap06.html#tagtcjh_3) and do not begin with a digit.
```

So we should say something along the lines of
```suggestion
CircleCI uses Bash, which follows the POSIX naming convention for environment variables. Valid characters include letters (uppercase and lowercase), digits, and the underscore. The first character of each environment variable must not be a digit.
```

This rule is in fact enforced in the CircleCI UI, I was not able to provision an environment variable name starting with a digit.

**Comment by HennaAbbas** [comment]
@jtatarik I tested this, I am able to create an env var that starts with a number.

**Comment by jtatarik** [comment]
> @jtatarik I tested this, I am able to create an env var that starts with a number.

@HennaAbbas - How? I'm getting, entirely as expected: 
Screenshot at 2024-09-30 17-08-25

**Comment by HennaAbbas** [comment]
I am able to add an env var that starts with a letter : 

Uploading Screen Recording 2024-09-30 at 10.20.53 AM.mov…

**Comment by HennaAbbas** [comment]
I am able to create an environment variable that starts with a number: 

https://github.com/user-attachments/assets/fc223290-5e00-4e82-ae46-73788ca7bd77

File: jekyll/_cci2/env-vars.md (hunk 1 above)
Comment (rosieyohannan) [reply jekyll/_cci2/env-vars.md:46]: Thank you @jtatarik! I just made a minor change to your suggestion

**Review by rosieyohannan — APPROVED** [review]

**Pull request merged**